### PR TITLE
Force progress support

### LINKED ISSUE #70: Add support for force progress
Add support for force progress, as described here https://github.com/aeternity/protocol/pull/462

### PR DESCRIPTION
**Pull request by Artur64**
resolves #70

### CODE REVIEW

File: apps/ae_socket_connector/lib/socket_connector.ex [hunk 1]
```diff
@@ -904,7 +951,8 @@ defmodule SocketConnector do
     "channels.sign.update",
     "channels.sign.initiator_sign",
     "channels.sign.deposit_tx",
-    "channels.sign.withdraw_tx"
+    "channels.sign.withdraw_tx",
+    "channels.sign.force_progress_tx"
```
Comment (AleksandarFilipov) [inline apps/ae_socket_connector/lib/socket_connector.ex:955]: Thinking that FP doesn't contain round. So it should be in the: `# Note, these doesn't contain round...` section

File: apps/ae_socket_connector/test/ae_socket_connector_test.exs [hunk 2]
```diff
@@ -1236,6 +1236,80 @@ defmodule SocketConnectorTest do
...
+                  initiator_contract,
+                  'make_move',
+                  ['11', '1']
+                )
+              end, :empty}
+         }},
+        {:initiator,
+         %{
+           # "consumed_forced_progress", comming as on_chain means, that FP was executed successfully
+           message: {:on_chain, "consumed_forced_progress"},
```
Comment (AleksandarFilipov) [inline apps/ae_socket_connector/test/ae_socket_connector_test.exs:1287]: would also expect, or is there a documentation missmatch?
```
{
  "jsonrpc": "2.0",
  "method": "channels.on_chain_tx",
  "params": {
    "channel_id": "ch_2eiGsAvt...",
    "data": {
      "info": "channel_changed",
      "tx": "tx_+QFxCwH4QrhA...",
      "type": "channel_force_progress_tx"
    }
  },
  "version": 1
}
```

File: apps/ae_socket_connector/lib/socket_connector.ex (hunk 1 above)
Comment (Artur64) [reply apps/ae_socket_connector/lib/socket_connector.ex:955]: This comes from FSM, the FSM prepares the transaction, updates and of course round is also a part of this. Then the user should like, take a look at everything and sign(agree) or disagree(not to sign) it. More info here: https://github.com/aeternity/protocol/blob/cb4dad7a5cba6c902f76c567cb6180cb4016ccf3/node/api/channel_ws_api.md#forcer-receives-a-prepared-force-progress-transaction-to-authenticate

Comment (Artur64) [reply apps/ae_socket_connector/lib/socket_connector.ex:955]: Sorry, misunderstood your comment, will be adjusted soon

File: apps/ae_socket_connector/test/ae_socket_connector_test.exs (hunk 2 above)
Comment (Artur64) [reply apps/ae_socket_connector/test/ae_socket_connector_test.exs:1287]: Yes, it should also expect `channels.on_chain_tx`, my fault, as I decided to suppress it...

File: apps/ae_socket_connector/lib/socket_connector.ex (hunk 1 above)
Comment (Artur64) [reply apps/ae_socket_connector/lib/socket_connector.ex:955]: Actually, after doing some research, it appears that the [`round`](https://github.com/aeternity/aeternity/blob/master/apps/aechannel/src/aesc_force_progress_tx.erl#L351) is in the `aesc_force_progress_tx`

File: apps/ae_socket_connector/test/ae_socket_connector_test.exs (hunk 2 above)
Comment (AleksandarFilipov) [reply apps/ae_socket_connector/test/ae_socket_connector_test.exs:1287]: Could we please add that to the test case for the sake of completeness

**Comment by Artur64** [comment]
Closing this now, as #97 has these changes integrated in it

**Pull request closed without merging**